source/documentation/guides/api_standards.html.md.erb

```diff
@@ -27,6 +27,21 @@ Adopt these standards if there is no reason not to, but you may break any of the
      - 2023-06-16T09:04:38Z
      - 09:04:46Z
 
+## Naming (REST)
+
+The primary data representation in REST is a resource, which is a mapping to a set of entities. This dictates how REST APIs should be designed and the following princples should guide how the resources are addressed:
+
+1. Resources are represented by nouns. Resource names can be "real" nouns or abstract collections e.g. `/users` or `/user-management`. 
+1. A resource can be a singleton or a collection. Collections are pluralised and singletons are addressed by ID. e.g. `/users/1`, where `users` is a collection and `1` is the singleton.
+1. Resources are hierarchical, with resources containing other resources. These relationships are denoted with forward slashes. e.g. `/teams/1/users/2`.
+1. A URI addresses the resource, not the action. HTTP methods should be used to indicate the action being performed. e.g. `HTTP POST /users`, not `/users/create`.
+  1. Where an action is required and doesn't fit the CRUD pattern, it should be supplied as a query parameter. e.g. `/scripts/1?action=execute`.
```

Comment:
I'm not a big fan of this for a couple of reasons:
- It makes it hard to define what a resource does, particularly in OpenAPI spec (since that's path-based and query strings aren't canonical elements, I believe)
- It leaves the method ambiguous: is it a PUT, a POST, a GET?
- I think it'd be easy to mess up since it would fallback to acting on the underlying resource. For example, if I `PUT /scripts/1?actin=execute` and don't notice the typo in "action", I could end up overwriting the script resource

An alternative for this action would be `POST /scripts/1/executions` since you're creating a new execution of the script, but I appreciate that might not translate well to other scenarios.

Reply:
Yeah I'm not a fan either really. I think this is the part where you have to say "not everything in a system will fit into the REST standard" and just accept that some actions on the system won't be addressing resources.

Do we want to try and set some standards here? e.g. if you are acting on a typically RESTful API, the requests should be POSTs? Or do we just want to say that we should strive to meet the standards but recognise that sometimes it is unavoidable?